### PR TITLE
fix access log body size

### PR DESCRIPTION
**Pull request by i110**
Currently, the access log `%b` specifier outputs *raw* body size ignoring compression, chunked encoding overhead, etc..
This PR fixes that issue.

### CODE REVIEW

**Comment by kazuho** [comment]
Thank you for the changes. Am I correct in assuming that Apache behaves the way proposed by this PR?

Specifically:
* does Apache count the overhead introduced by chunked encoding in `%b`?
* does Apache log the number of bytes after compression when mod_deflate is used?

**Comment by i110** [comment]
@kazuho I checked Apache's behavior

> does Apache count the overhead introduced by chunked encoding in %b?

No.

> does Apache log the number of bytes after compression when mod_deflate is used?

Yes.

So at https://github.com/h2o/h2o/pull/1478/commits/69f47636fb51948431a682c31d260ec3c56f3b9e I made bytes_send not contain the overhead of chunked encoding.

File: lib/http1.c [hunk 1]
```diff
@@ -707,10 +710,21 @@ void finalostream_send(h2o_ostream_t *_self, h2o_req_t *req, h2o_iovec_t *inbufs
     struct st_h2o_http1_finalostream_t *self = (void *)_self;
     struct st_h2o_http1_conn_t *conn = (struct st_h2o_http1_conn_t *)req->conn;
     h2o_iovec_t *bufs = alloca(sizeof(h2o_iovec_t) * (inbufcnt + 1));
+    int i;
     int bufcnt = 0;
 
     assert(self == &conn->_ostr_final);
 
+    /* count bytes_sent if other ostreams haven't counted */
+    if (inbufcnt > 0 && req->bytes_sent == 0) {
```
Comment (kazuho) [inline lib/http1.c]: I can understand the fact that you wanted to keep modifications within http1.c, but isn't this a bit complicated?

How about adding a flag to `h2o_req_t` that tells if the bytes are counted outside of the protocol handler (and set the flag in chunked.c)?

We already have HTTP/1 specific flag in `h2o_req_t` (as `http1_is_persistent`). I think we can add another.

**Comment by i110** [comment]
@kazuho fixed it!

**Comment by kazuho** [comment]
Thank you for the PR!

**Pull request merged**